controllers/component_build_controller_pac.go

```diff
@@ -419,6 +443,12 @@ func (r *ComponentBuildReconciler) ensurePaCRepository(ctx context.Context, comp
 				return err
 			}
 			if err := r.Client.Create(ctx, repository); err != nil {
+				if strings.Contains(err.Error(), "repository already exist with url") {
```

Comment:
Above code has been handled the multi-components case, and here the component should not have a PaC Repository yet. Why check this error again here, would it happen again in some cases?